File: scala/src/main/scala/com/laamella/code_state_machine/condition/SingleEventMatchCondition.scala
package com.laamella.code_state_machine.condition

/**
 * This condition is met when the event is equal to the event passed in the
 * constructor.
 */
final class SingleEventMatchCondition[E](singleEvent: E) extends EventBasedCondition[E] {

  override def toString = s"is $singleEvent"

  override protected def conditionIsMetAfterHandlingEvent(event: E) = singleEvent equals event
}

File: java/src/main/java/com/laamella/code_state_machine/condition/SingleEventMatchCondition.java
package com.laamella.code_state_machine.condition;

/**
 * This condition is met when the event is equal to the event passed in the
 * constructor.
 */
public final class SingleEventMatchCondition<E> extends EventBasedCondition<E> {
    private final E singleEvent;

    public SingleEventMatchCondition(E singleEvent) {
        this.singleEvent = singleEvent;
    }

    @Override
    public String toString() {
        return "is " + singleEvent;
    }

    @Override
    protected boolean conditionIsMetAfterHandlingEvent(E event) {
        return singleEvent.equals(event);
    }
}